source/extensions/filters/listener/original_dst/original_dst.cc

```diff
@@ -20,12 +22,47 @@ Network::FilterStatus OriginalDstFilter::onAccept(Network::ListenerFilterCallbac
 
   if (socket.addressType() == Network::Address::Type::Ip) {
     Network::Address::InstanceConstSharedPtr original_local_address = getOriginalDst(socket);
-
     // A listener that has the use_original_dst flag set to true can still receive
     // connections that are NOT redirected using iptables. If a connection was not redirected,
     // the address returned by getOriginalDst() matches the local address of the new socket.
     // In this case the listener handles the connection directly and does not hand it off.
     if (original_local_address) {
+#ifdef WIN32
+      // See how to perform bind or connect redirection on MSDN
+      // https://docs.microsoft.com/en-us/windows-hardware/drivers/network/using-bind-or-connect-redirection
+      if constexpr (Network::win32SupportsOriginalDestination()) {
+        if (traffic_direction_ == envoy::config::core::v3::OUTBOUND) {
+          ENVOY_LOG(debug, "[Windows] Querying for redirect record for outbound listener");
```

Comment:
consider moving the log statement to after the Ioctl call so you can always log the returned status and maybe even the redirect record.